### PR TITLE
feat: Add slideshow captions to Fronts

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Renders captions in twirl templates on Fronts.

Depends on #24791
Depends on #24781 
Part of #24782

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before]: https://user-images.githubusercontent.com/21217225/159292921-50fbe146-2de8-4881-857f-819c65a8a9f9.png
[after]: https://user-images.githubusercontent.com/21217225/159292834-604a4116-df33-4d73-b2eb-d596d1949f16.png

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [x] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [x] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

#### Plain white background 
image

#### Max length caption

image

#### Mobile image

image

#### Max length caption on Mobile

image

### CODE REVIEW

File: static/src/stylesheets/module/facia-garnett/_item.scss [hunk 1]
```diff
@@ -540,24 +540,37 @@ $block-height: 58px;
...
     }
-}
 
+    figcaption {
+        @include fs-bodyHeading(1);
+        position: absolute;
+        bottom: 0;
+        left: 0;
+        width: 100%;
+        background: linear-gradient(180deg, rgba(0, 0, 0, 0) 0%,  rgba(0, 0, 0, .8) 100%);
```
Comment (mxdvl) [inline static/src/stylesheets/module/facia-garnett/_item.scss:569]: Suggestion: `to bottom`

```suggestion
        background: linear-gradient(to bottom, rgba(0, 0, 0, 0) 0%,  rgba(0, 0, 0, .8) 100%);
```

**Review by mxdvl — COMMENTED** [review]
what’s the difference between `facia` and `facia-garnett`? Do both need to be updated?

**Comment by AshCorr** [comment]
> what’s the difference between facia and facia-garnett? Do both need to be updated?

I'd love to know that myself! They had identical sections for Slideshows before so I updated both, but would be good to find out why this CSS is duplicated.

**Comment by mxdvl** [comment]
Yes, _I think_ Garnett is the new version of fronts.

**Review by jamesgorrie — COMMENTED** [review]
Looking great!

File: static/src/stylesheets/module/facia-garnett/_item.scss [hunk 2]
```diff
@@ -540,24 +540,37 @@ $block-height: 58px;
...
             animation-timing-function: linear;
             animation-iteration-count: infinite;
             animation-direction: normal;
             opacity: 0;
         }
     }
-}
 
+    figcaption {
+        @include fs-bodyHeading(1);
```
Comment (AshCorr) [inline static/src/stylesheets/module/facia-garnett/_item.scss:564]: This needs to be a smaller size on mobile.

**Review by mxdvl — APPROVED** [review]

**Comment by SiAdcock** [comment]
> Garnett

Garnett was the codename of a project 5 years ago to [restyle the entire website and apps](https://github.com/guardian/frontend/pulls?q=is%3Apr+garnett+is%3Aclosed+sort%3Acreated-asc). It was done fast, and in a big bang manner, for reasons I can discuss over beers 🍻

We expected to clean up the code after it was implemented, but in some cases this has evaded us. It is unclear to what extent unused, pre-Garnett code still exists and can just be deleted. Perhaps something to raise in the Dotcom Alumni chat.

**Review by OllysCoding — APPROVED** [review]
+1 This looks really good! Is the experiment something that cane be opted in and out of individually?

**Comment by AshCorr** [comment]
> This looks really good! Is the experiment something that cane be opted in and out of individually?

Yep, @jamesgorrie created a slideshow-captions experiment for this. Its set to 0% but I guess we can increase it once the fronts tool is updated and everyone is happy with what we have.

**Pull request merged**